drivers/sensor/nxp/mcux_lpcmp/mcux_lpcmp.c

```diff
@@ -0,0 +1,434 @@
+
+	if ((int16_t)trig->chan != SENSOR_CHAN_MCUX_LPCMP_OUTPUT) {
+		return -ENOTSUP;
+	}
+
+	if ((int16_t)trig->type == SENSOR_TRIG_MCUX_LPCMP_OUTPUT_RISING) {
+		data->rising_handler = handler;
+		data->rising_trigger.type = trig->type;
+		data->rising_trigger.chan = trig->chan;
+	} else if ((int16_t)trig->type == SENSOR_TRIG_MCUX_LPCMP_OUTPUT_FALLING) {
+		data->falling_handler = handler;
+		data->falling_trigger.type = trig->type;
+		data->falling_trigger.chan = trig->chan;
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static void mcux_lpcmp_trigger_work_handler(struct k_work *item)
+{
+	struct mcux_lpcmp_data *data = CONTAINER_OF(item, struct mcux_lpcmp_data, work);
+	struct sensor_trigger trigger;
+	sensor_trigger_handler_t handler = NULL;
+
+	if (((data->status & kLPCMP_OutputRisingEventFlag) == kLPCMP_OutputRisingEventFlag) &&
+	    ((data->status & kLPCMP_OutputAssertEventFlag) == kLPCMP_OutputAssertEventFlag)) {
+		trigger = data->rising_trigger;
+		handler = data->rising_handler;
+	} else if (((data->status & kLPCMP_OutputFallingEventFlag) ==
+		    kLPCMP_OutputFallingEventFlag) &&
+		   ((data->status & kLPCMP_OutputAssertEventFlag) !=
+		    kLPCMP_OutputAssertEventFlag)) {
+		trigger = data->falling_trigger;
+		handler = data->falling_handler;
+	} else {
+		return;
+	}
+
+	if (handler) {
+		handler(data->dev, &trigger);
+	}
+}
+#endif /* CONFIG_MCUX_LPCMP_TRIGGER */
+
+static int mcux_lpcmp_init(const struct device *dev)
+{
+	const struct mcux_lpcmp_config *config = dev->config;
+	struct mcux_lpcmp_data *data = dev->data;
+	int err;
+
+	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+	if (err) {
+		return err;
+	}
+
+	/* LPCMP configuration */
+	LPCMP_GetDefaultConfig(&data->lpcmp_config);
+	data->lpcmp_config.powerMode = config->power_level;
+	data->lpcmp_config.hysteresisMode = config->hysteresis_level;
+	data->lpcmp_config.enableOutputPin = config->output_enable;
+	data->lpcmp_config.enableInvertOutput = config->output_invert;
+	data->lpcmp_config.useUnfilteredOutput = config->unfiltered;
+#if defined(FSL_FEATURE_LPCMP_HAS_CCR1_FUNC_CLK_SEL) && FSL_FEATURE_LPCMP_HAS_CCR1_FUNC_CLK_SEL
+	data->lpcmp_config.functionalSourceClock = config->function_clock;
+#endif /* FSL_FEATURE_LPCMP_HAS_CCR1_FUNC_CLK_SEL */
+	LPCMP_Init(config->base, &data->lpcmp_config);
+
+#ifdef CONFIG_MCUX_LPCMP_TRIGGER
+	data->dev = dev;
+	k_work_init(&data->work, mcux_lpcmp_trigger_work_handler);
+	config->irq_config_func(dev);
+	LPCMP_EnableInterrupts(config->base, kLPCMP_OutputRisingInterruptEnable |
+						     kLPCMP_OutputFallingInterruptEnable);
+#endif /* CONFIG_MCUX_LPCMP_TRIGGER */
+
+	LPCMP_Enable(config->base, true);
+
+	return 0;
+}
+
+static const struct sensor_driver_api mcux_lpcmp_driver_api = {
+	.attr_set = mcux_lpcmp_attr_set,
+	.attr_get = mcux_lpcmp_attr_get,
+#ifdef CONFIG_MCUX_LPCMP_TRIGGER
+	.trigger_set = mcux_lpcmp_trigger_set,
+#endif /* CONFIG_MCUX_LPCMP_TRIGGER */
+	.sample_fetch = mcux_lpcmp_sample_fetch,
```

Comment:
Where is the `channel_get` implementation? It isn't clear to me how the user is supposed to read the LPCMP value if they aren't using triggering.

Reply:
The user could get the lpcmp output through fetch function.

Reply:
that is not how the sensor api is written

Reply:
@decsny is correct here. The `sensor_sample_fetch` function (per the [documentation](https://docs.zephyrproject.org/latest/hardware/peripherals/sensor.html#c.sensor_sample_fetch)) simply instructs the driver to fetch a sample and store it into an internal buffer. The `sensor_channel_get` function (which is the implementation this driver needs to provide) is how the user accesses the data in that internal buffer. Please take a look at the MCUX ACMP driver for an example of how this should be implemented: https://github.com/zephyrproject-rtos/zephyr/blob/a1fc3b78efefdbaca79ae239d2615fae437d4fc1/drivers/sensor/nxp/mcux_acmp/mcux_acmp.c#L351

Reply:
Yes, as I checked, the **channel_get** function must be provided, I will add it in the next update. Thank you everyone for the reminder.